lms/static/js/student_account/enrollment.js

```diff
@@ -36,6 +36,17 @@
                         // If so, redirect to a page explaining to the user
                         // why they were blocked.
                         this.redirect(responseData.user_message_url);
+                    } else if (jqXHR.status === 400) {
+                        // Show the error message for bad requests (invalid enrollment data)
+                        this.showMessage(responseData);
+                        // Add timeout before redirecting to allow user to see the message
+                        var self = this;
```

Comment:
```suggestion
                        const self = this;
```
However, I think you can avoid this definition using arrow function according with what is described here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this#description, arrow function works with the `this` that comes from the surrounding context.
```suggestion
setTimeout(() => {
    if (redirectUrl) {
        this.redirect(redirectUrl);
    }
}, 6000);
```

I couldn't configure my local environment for testing, for testing this specific code, but you can try.

Reply:
I have the environment settled, I'll try it!

Reply:
Done !